drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1218 @@
+
+	/* calc remaining part (first mT digit + fractal part) and scale according to Zephyr.
+	 * Ensure that always positive.
+	 */
+	const int64_t raw_dec_part = (b_field->val1 * (1 << 16)) / CONV_FACTOR_MT_TO_GS;
+
+	b_field->val2 = ((raw_value - raw_dec_part) * 1000000 * CONV_FACTOR_MT_TO_GS) / (1 << 16);
+
+	if ((b_field->val1 != 0) && (b_field->val2 < 0)) {
+		/* negative fractional part only make sense if integer part is zero */
+		b_field->val2 = -b_field->val2;
+	}
+}
+
+/**
+ * @brief calculates the temperature value
+ *
+ * @param[in] raw_value data read from the device
+ * @param[out] temperature holds the result data after the operation
+ */
+static inline void _tmag5273_temperature_convert(int64_t raw_value,
+					struct sensor_value *temperature)
+{
+	const int64_t value = TMAG5273_TEMPERATURE_T_SENS_T0
+			+ ((raw_value - TMAG5273_TEMPERATURE_T_ADC_T0)
+				/ TMAG5273_TEMPERATURE_T_ADC_RES)
+			* 1000000;
+
+	temperature->val1 = value / 1000000;
+	temperature->val2 = value % 1000000;
+}
+
+/**
+ * @brief calculates the angle value between two axis
+ *
+ * @param[in] raw_value data read from the device
+ * @param[out] angle holds the result data after the operation
+ */
+static inline void _tmag5273_angle_convert(int16_t raw_value, struct sensor_value *angle)
+{
+	angle->val1 = (raw_value >> 4) & 0x1FF;
+	angle->val2 = ((raw_value & 0xF) * 1000000) >> 1;
+}
+
+/**
+ * @brief calculates the magnitude value in G between two axis
+ *
+ * Note that \c MAGNITUDE_RESULT represents the MSB of the calculation,
+ * therefore it needs to be shifted.
+ *
+ * @param[in] raw_value data read from the device
+ * @param[out] magnitude holds the result data after the operation
+ */
+static inline void _tmag5273_magnitude_convert(uint8_t raw_value, const uint16_t range,
+			struct sensor_value *magnitude)
+{
+	_tmag5273_channel_b_field_convert(raw_value << 8, range, magnitude);
+}
+
+static int tmag5273_channel_get(const struct device *dev,
+				enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	if (val == NULL) {
+		LOG_DBG("val: NULL");
+		return -EINVAL;
+	}
+
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+
+	int8_t val_offset = 0;
+
+	{
+		const bool all_mag_axis = (chan == SENSOR_CHAN_MAGN_XYZ)
+					|| (chan == SENSOR_CHAN_ALL);
+
+		if (drv_cfg->x_axis && (all_mag_axis || (chan == SENSOR_CHAN_MAGN_X))) {
+			_tmag5273_channel_b_field_convert(drv_data->x_sample, drv_data->xyz_range,
+				val + val_offset);
+			val_offset++;
+		}
+
+		if (drv_cfg->y_axis && (all_mag_axis || (chan == SENSOR_CHAN_MAGN_Y))) {
+			_tmag5273_channel_b_field_convert(drv_data->y_sample, drv_data->xyz_range,
+				val + val_offset);
+			val_offset++;
+		}
+
+		if (drv_cfg->z_axis && (all_mag_axis || (chan == SENSOR_CHAN_MAGN_Z))) {
+			_tmag5273_channel_b_field_convert(drv_data->z_sample, drv_data->xyz_range,
+				val + val_offset);
+			val_offset++;
+		}
+	}
```

Comment:
Why is this block enclosed in braces?

Reply:
Same as above in the init function. It's more out of a habit. If this is not desired here, I'll change it.